src/runtime/hexagon_remote/remote.cpp

```diff
@@ -44,15 +95,239 @@ int halide_do_task(void *user_context, halide_task_t f, int idx,
+            // and tries to do it. If it cannot find a job, it goes
+            // to sleep.
+            if (num_jobs > 0) {
+                for (job_idx = MAX_WORKLOADS-1 ; job_idx >= 0; job_idx--) {
+                    if (!jobs[job_idx].claimed()) {
+                        myjob = jobs[job_idx];
+                        orig_job = &jobs[job_idx];
+                        job_found = true;
+                        break;
+                    }
+                }
+            }
+            if (!job_found) {
+                // release the lock and go to sleep.
+                // ***************************
+                // *** Work queue unlocked ***
+                // ***************************
+                FARF(LOW, "HVX_TP: %d: goto_work: Couldn't find a job, going to sleep\n", tid);
+                unlock_work_queue();
+                qurt_sem_down(&wait_for_work);
+                FARF(LOW, "HVX_TP: %d: goto_work: Got woken up. Going to look for work.\n", tid);
+                 // ***********************
+                // *** Lock work queue ***
+                // ***********************
+                lock_work_queue();
+                continue;
+            } else {
+                FARF(LOW, "HVX_TP: %d: goto_work: Found a job at idx = %d working on x = %d\n", tid, job_idx, myjob.next);
+            }
+        } else {
+            // We are here only if this thread owns a job that is not done.
+            // So do a part of it.
+            if (owned_job->claimed()) {
+                // the owner goes to sleep till the workers wake it up.
+                FARF(LOW, "HVX_TP: %d: goto_work: Owner about to sleep\n", tid);
+                qurt_cond_wait(&owned_job->wakeup_owner, &work_queue.work_mutex);
+                FARF(LOW, "HVX_TP: %d: goto_work: Owner waking up.\n", tid);
+                // This thread should have the lock now after having been woken up.
+                break;
+            } else {
+                myjob = *owned_job;
+                orig_job = owned_job;
+                FARF(LOW, "HVX_TP: %d: goto_work: Owner about to work\n", tid);
+            }
+        }
+        orig_job->next++;
+        orig_job->active_workers++;
+
+        // ***************************
+        // *** Work queue unlocked ***
+        // ***************************
+        unlock_work_queue();
+
+        FARF(LOW, "HVX_TP: %d: goto_work: About to do_task, user_context = 0x%x, f = 0x%x x = %d \n", tid, myjob.user_context, myjob.f, myjob.next);
+        int result = halide_do_task(myjob.user_context, myjob.f,
+                                    myjob.next, myjob.closure);
+        FARF(LOW, "HVX_TP: %d: goto_work: Finished do_task with status = %d\n", tid, result);
+        // ***********************
+        // *** Lock work queue ***
+        // ***********************
+        lock_work_queue();
+        if (result) {
+            orig_job->exit_status = result;
```

Comment:
If there was an error, it should bail out of the rest of the jobs, it looks like this will continue to work on the job? Ctrl+F "
"exit_status" shows it is unused except for initialization and here BTW. It's probably not getting returned anywhere.